### PR TITLE
Feature/merging

### PR DESCRIPTION
**Pull request by keiranmraine**
## Primary changes:

1. Dual-guide count file format updated to match that of single-guide (which it should have to start with).
2. Merging of counts files implemented.
3. Merging of `stats.json` performed as part of count merging.

Merging will not allow identical files to be merged.

## Merged count file

The header is of the following format.  `...` indicates manual truncation for brevity:

```
##Command: pycroquet merge-counts -o merge-dg -i dg_output.counts.tsv.gz -i dg_output.counts.tsv.gz --low_count 50
##Version: 1.3.0
##Count-col-#1: md5: XXXXXXXXXXX; Version: 1.3.0; Command: pycroquet dual-guide ...
##Count-col-#2: md5: YYYYYYYYYYY; Version: 1.3.0; Command: pycroquet dual-guide ...
#id     sgrna_ids       sgrna_seqs      gene_pair_id    unique_guide    reads_SAMPLENAME   1       2
```

The `Count-col-#N` headers capturing the count file md5 (or sha256 depending on choice) command/version used to generate the original count is correlated to the number columns following the `reads_SAMPLENAME` total column.

## Stats json file

This has been extended to include a `merged_from` section to allow the embedding of the individual count stats (which includes original command and version also).  The order of the json list matches the `Count-col-#N` headers in the count file.

### CODE REVIEW

File: pycroquet/cli.py [hunk 1]
```diff
@@ -291,7 +292,7 @@ def dual_guide(*args, **kwargs):
 @debug_params
 def long_read(*args, **kwargs):
     """
-    log-read - map read file to library guides and output counts, statistics and alignments files.
+    Map read file to library guides and output counts, statistics and alignments files.
```
Comment (vaofford) [inline pycroquet/cli.py:295]: Do we need to take into account the library-independent quantification that was built in with the `unique_only` option for the description? i.e. we don't always map to the library? @keiranmraine

File: pycroquet/merge.py [hunk 2]
```diff
@@ -27,12 +27,214 @@
...
+            items = line.split("\t")
+            data_set.append([*items[0:5], int(items[-1])])
+
+        line = i_fh.readline()
+    i_fh.close()
+
+    return (merge_header_line(cmd, version, file_idx, chk_item), sample, data_set, chk_item)
+
+
+def merge_count_data(inputs: List[str], chksum_type: str):
```
Comment (vaofford) [inline pycroquet/merge.py:149]: What's the expected behaviour if someone submits a single file for merging?

Comment (vaofford) [reply pycroquet/merge.py:149]: Ignore, covered/answered in next file.

**Review by vaofford — COMMENTED** [review]
One question on whether we need to broaden descriptions to account for library-independent , unique-only mode?

File: pycroquet/cli.py (hunk 1 above)
Comment (keiranmraine) [reply pycroquet/cli.py:295]: Hmm, how to describe as part of that is a little tricky.  I wonder if we should just create a separate sub-command, thoughts (and idea what to call it).

Comment (keiranmraine) [reply pycroquet/cli.py:295]: How about (if not adding an additional sub-command):

> Map read file to library guides and output counts, statistics and alignments files.  Minimal assessment of unique reads only via `--unique` option.

**Review by vaofford — APPROVED** [review]

**Pull request merged**